### PR TITLE
feat(icons): added `calendar-sync` icon

### LINKED ISSUE #2071: [icon request] calendar-repeat
### Icon name

calendar-repeat

### Use cases

I need this icon to use it for `recurring events` like for example daily or weekly discount events.
it also
Here are some more use cases for a recurring events icon, in addition to daily or weekly discount events:
* **Cron jobs scheduling:** This could be scheduling a recurring job to handle invalid urls in database.
* **Recurring Events or Limited time offers or promotions:**  This could be a flash sale, a seasonal promotion, or a free trial offer.
* **Tasks or reminders that need to be repeated:** This could be taking medication, watering plants, or walking the dog.
* **Loan auto periodic payment** This could be for handling loan installments as reminder or auto.

### Design ideas

it can be done with combining the `calendar` icon and `repeat-2` icon. I wanted to do it myself but had some problem with size of both icons so I made this request.

somehow related to this [issue #1427](https://github.com/lucide-icons/lucide/issues/1427)

### Checklist

- [X] I have searched if someone has submitted a similar issue before and there weren't any. (Please make sure to also search closed issues, as this issue might already have been resolved.)
- [X] I have searched existing icons to make sure it does not already exist and I didn't find any.
- [X] I am not requesting a brand logo and the art is not protected by copyright.
- [X] I am not requesting an icon that includes religious, political imagery or hate symbols.
- [X] I have provided appropriate use cases for the icon(s) requested.

### PR DESCRIPTION
**Pull request by chessurisme**
close #2071
## What is the purpose of this pull request?
- [x] New Icon

### Description
Added new `calendar-sync` icon.

### Icon use case

Here are some more use cases:

- Cron jobs scheduling: This could be scheduling a recurring job to handle invalid urls in database.
Recurring Events or Limited time offers or promotions: This could be a flash sale, a seasonal promotion, or a free trial offer.
- Tasks or reminders that need to be repeated: This could be taking medication, watering plants, or walking the dog.
- Loan auto periodic payment This could be for handling loan installments as reminder or auto.

### Concept

- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [ ] The icons are solely my own creation.
- [x] The icons were originally created in #<issueNumber> by @danielbayley
- [x] I've based them on the following Lucide icons: `folder-sync`, `calendar-heart`
- [ ] I've based them on the following design: 

### Naming

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to three points of precision.

## Before Submitting
- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

**Comment by chessurisme** [comment]
Everytime I pushed I have this annoying four first commits. I need to troubleshoot.

**Comment by jguddas** [comment]
> Everytime I pushed I have this annoying four first commits. I need to troubleshoot.

You just have random stuff in your main branch, just have to force remove it to be the same as the lucide branch.

**Review by jguddas — CHANGES_REQUESTED** [review]
Has box-select removed in the PR.

**Comment by chessurisme** [comment]
it is already removed

**Comment by karsa-mistmere** [comment]
The sync arrows are .5px off grid vertically.

**Comment by chessurisme** [comment]
> The sync arrows are .5px off grid vertically.

Already fixed!

**Comment by chessurisme** [comment]
Time to go! 🚀

File: icons/calendar-sync.svg [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M12 10.5v4h4" />
+  <path d="m12 14.5 1.535-1.605a5 5 0 0 1 8 1.5" />
+  <path d="M16 2v4" />
+  <path d="M21 7.5V6a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h3.5" />
```
Comment (jguddas) [inline icons/calendar-sync.svg:15]:
```suggestion
  <path d="M21 8.5V6a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h5" />
```

<a title="Open lucide studio" target="_blank" href="https://lucide-studio.vercel.app/edit?value=%3Cpath+d=%22M12+10.5v4h4%22+/%3E%3Cpath+d=%22m12+14.5+1.535-1.605a5+5+0+0+1+8+1.5%22+/%3E%3Cpath+d=%22M16+2v4%22+/%3E%3Cpath+d=%22M21+8.5V6a2+2+0+0+0-2-2H5a2+2+0+0+0-2+2v14a2+2+0+0+0+2+2h5%22+/%3E%3Cpath+d=%22m22+18.5-1.535+1.605a5+5+0+0+1-8-1.5%22+/%3E%3Cpath+d=%22M22+22v-4h-4%22+/%3E%3Cpath+d=%22M3+10h5%22+/%3E%3Cpath+d=%22M8+2v4%22+/%3E">icons<br/>Open lucide studio</a>

**Review by jguddas — CHANGES_REQUESTED** [review]

**Review by jguddas — CHANGES_REQUESTED** [review]
The refresh icon has a 2px gap violation and is not sympathetical

**Comment by chessurisme** [comment]
How about this:
<a title="Open lucide studio" target="_blank" href="https://lucide-studio.vercel.app/edit?value=%3Cpath+d=%22M11+10.5v4h4%22+/%3E%3Cpath+d=%22m11+14.5+1.535-1.605a5+5+0+0+1+8+1.5%22+/%3E%3Cpath+d=%22M16+2v4%22+/%3E%3Cpath+d=%22m21+18.5-1.535+1.605a5+5+0+0+1-8-1.5%22+/%3E%3Cpath+d=%22M21+22v-4h-4%22+/%3E%3Cpath+d=%22M21+9V6a2+2+0+0+0-2-2H5a2+2+0+0+0-2+2v14a2+2+0+0+0+2+2h4%22+/%3E%3Cpath+d=%22M3+10h4%22+/%3E%3Cpath+d=%22M8+2v4%22+/%3E">icons<br/>Open lucide studio</a>

**Comment by jguddas** [comment]
> How about this, I moved the refresh icon so there will be 2px on each sides: [iconsOpen lucide studio](https://lucide-studio.vercel.app/edit?value=%3Cpath+d=%22M11+10v4h4%22+/%3E%3Cpath+d=%22m11+14+1.535-1.605a5+5+0+0+1+8+1.5%22+/%3E%3Cpath+d=%22M16+2v4%22+/%3E%3Cpath+d=%22m21+18-1.535+1.605a5+5+0+0+1-8-1.5%22+/%3E%3Cpath+d=%22M21+22v-4h-4%22+/%3E%3Cpath+d=%22M21+8.5V6a2+2+0+0+0-2-2H5a2+2+0+0+0-2+2v14a2+2+0+0+0+2+2h4%22+/%3E%3Cpath+d=%22M3+10h4%22+/%3E%3Cpath+d=%22M8+2v4%22+/%3E)

That looks pretty good, we should closed the gap at the bottom like this:

<a title="Open lucide studio" target="_blank" href="https://lucide-studio.vercel.app/edit?value=%3Cpath+d=%22M11+10v4h4%22+/%3E%3Cpath+d=%22m11+14+1.535-1.605a5+5+0+0+1+8+1.5%22+/%3E%3Cpath+d=%22M16+2v4%22+/%3E%3Cpath+d=%22m21+18-1.535+1.605a5+5+0+0+1-8-1.5%22+/%3E%3Cpath+d=%22M21+22v-4h-4%22+/%3E%3Cpath+d=%22M21+8.5V6a2+2+0+0+0-2-2H5a2+2+0+0+0-2+2v14a2+2+0+0+0+2+2h4.3%22+/%3E%3Cpath+d=%22M3+10h4%22+/%3E%3Cpath+d=%22M8+2v4%22+/%3E">icons<br/>Open lucide studio</a>

**Comment by chessurisme** [comment]
I also think about that...

**Review by jguddas — APPROVED** [review]

**Review by ericfennis — APPROVED** [review]

**Pull request merged**